### DISCUSSION TITLE
File explorer fails to show contents of mounted directories

### DISCUSSION THREAD

**Discussion by Bongani** [coder/code-server#2034] [Q&A]
Hi,

I recently attempted to upgrade to code-server 3.5.0 (and before this to code-server 3.4.1), from code-server version 2.1692-vsc1.39.2. With both versions I had picked that my mounted (fusermount) directories where available in the explorer, but when I clicked the directory it did not show the contents of the folder. I was however able to run `ls` in the terminal and see the contents of the mounted directory. There was no error in the logs why the mounted directory did not show the contents.

A bit of investigation today relieved a bit more when I ran `ls -la` on the mounted directory.

![image (6)](https://user-images.githubusercontent.com/1842813/91724588-cecaa000-eb9d-11ea-8d72-9b22413da1a5.png)

As you can see from the screenshot, there is folder/directory that appears to be at fault. Once I ran `mkdir tf`, the VS Code explorer was able to show the contents of the mounted directory. So I guess in the older version of code-server, the explorer was failing silently and not showing the faulty folder `tf`, but now with newer releases the entire mounted directory is failing to show.

My first question would be, is there a property I can set that will ignore future cases were code-server can fail silently on faulty folders and show the entire mounted directory?  If not, can this be reported as bug instead in order to get it fix?

**Comment by code-asher** [comment]
Do you have any idea how the `tf` entry was created? If we can recreate it that'll help us narrow down the bug.

**Comment by Bongani** [comment]
Hi @code-asher ,

I believe that `tf` folder was created by tensorflow, while running a notebook.

I do have more information though. It appears that it was a mounting issue from [s3fs-fuse](https://github.com/s3fs-fuse/s3fs-fuse) which caused some folders to appear that way. We switched to [goofys](https://github.com/kahing/goofys) for some testing and no longer get that issue were a folder/directory has `??????` with some folders. Now the VS Code explorer with codes-server 3.5.0 can show the directories.

I'm not sure if there is a way to replicate creating a folder with `???????`, if you know of one I can try test it out on my side.

**Reply by code-asher** [reply]
I'm not sure how to replicate it but I'll poke around the explorer code and see if there's anything that could cause issues. My guess is if something fails to `stat` then the whole explorer just gives up or something like that.

**Reply by Bongani** [reply]
@code-asher did you have any luck? Even though goofys seemed like a good filey system replacement it overwhelmed our system, so we had to fall back to s3fs-fuse. And now stuck at the same problem again.

**Reply by code-asher** [reply]
It's on my list but I haven't had a chance yet. Does the problem persist
with 3.6.0?
